subsys/bluetooth/host/gatt.c

```diff
@@ -5210,6 +5227,71 @@ int bt_gatt_store_ccc(uint8_t id, const bt_addr_le_t *addr)
 	return 0;
 }
 
+/** Struct used to store both the id and the random address of a device when replacing
+ * random addresses in the ccc attribute's cfg array with the device's id address after
+ * pairing complete.
+ */
+struct addr_match {
+	const bt_addr_le_t *private_addr;
+	const bt_addr_le_t *id_addr;
+};
+
+uint8_t convert_to_id_on_match(const struct bt_gatt_attr *attr,
+							   uint16_t handle, void *user_data)
+{
+	struct _bt_gatt_ccc *ccc;
+	struct addr_match *device_addresses = user_data;
+
+	/* Check if attribute is a CCC */
+	if (attr->write != bt_gatt_attr_write_ccc) {
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	ccc = attr->user_data;
+
+	/* Copy the device's id address to the config's address if the config's address is the
+	 * same as the device's private address
+	 */
+	for (size_t i = 0; i < ARRAY_SIZE(ccc->cfg); i++) {
+		if (bt_addr_le_cmp((const bt_addr_le_t *)&ccc->cfg[i].peer,
```

Comment:
The `(const bt_addr_le_t *)` typecast seems unnecessary.

Reply:
Removed the typecast.